virtcontainers/network.go

```diff
@@ -579,6 +580,111 @@ func newNetwork(networkType NetworkModel) network {
 	}
 }
 
+const procMountInfoFile = "/proc/self/mountinfo"
+
+// getNetNsFromBindMount returns the network namespace for the bind-mounted path
+func getNetNsFromBindMount(nsPath string, procMountFile string) (string, error) {
+	netNsMountType := "nsfs"
+
+	// Resolve all symlinks in the path as the mountinfo file contains
+	// resolved paths.
+	resolvedPath, err := filepath.EvalSymlinks(nsPath)
+	if err != nil {
+		return "", err
+	}
+	nsPath = resolvedPath
+
+	f, err := os.Open(procMountFile)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		text := scanner.Text()
+
+		// Scan the mountinfo file to search for the network namespace path
+		// This file contains mounts in the eg format:
+		// "711 26 0:3 net:[4026532009] /run/docker/netns/default rw shared:535 - nsfs nsfs rw"
+		//
+		// Reference: https://www.kernel.org/doc/Documentation/filesystems/proc.txt
+
+		// We are interested in the first 9 fields of this file,
+		// to check for the correct mount type.
+		fields := strings.Split(text, " ")
+		if len(fields) < 9 {
+			fmt.Printf("Found fields less than 9\n")
+			continue
+		}
+
+		index := strings.Index(text, " - ")
+		if index < 0 {
+			fmt.Printf("Could not find delimiter\n")
+			continue
+		}
+
+		// Field following the separator contains the mount type
+		// We check here if the mount type is a network namespace mount type, namely "nsfs"
+
+		postSeparatorFields := strings.Fields(text[index+3:])
+		if postSeparatorFields[0] != netNsMountType {
+			fmt.Printf("Could not match network namespace mount type\n")
+			continue
+		}
+
+		// This is the mount point/destination for the mount
+		if fields[4] != nsPath {
+			fmt.Printf("Did not match the network ns path provided : %s\n", fields[4])
+			continue
+		}
+
+		// This is the root/source of the mount
+		fmt.Printf("Returning result : %s\n", fields[3])
+		return fields[3], nil
+	}
+
+	return "", nil
+}
+
+// hostNetworkingRequested checks if the network namespace requested is the
+// same as the current process.
+func hostNetworkingRequested(configNetNs string) bool {
+	var evalNS, nsPath, currentNsPath string
+	var err error
+
+	// Net namespace provided as "/proc/pid/ns/net"
+	if strings.HasPrefix(configNetNs, "/proc") && strings.HasSuffix(configNetNs, "/ns/net") {
+		// Here we are trying to resolve the path but it fails because
+		// namespaces links don't really exist. For this reason, the
+		// call to EvalSymlinks will fail when it will try to stat the
+		// resolved path found. As we only care about the path, we can
+		// retrieve it from the PathError structure.
+
+		if _, err = filepath.EvalSymlinks(configNetNs); err != nil {
```

Comment:
What if the `EvalSymlinks()` returns `err = nil` ? Should we simply return false ? Or return an error ?